### ISSUE TITLE
Discussion: Batched Queries

### ISSUE THREAD

**Issue by devknoll** [facebook/relay#132]
Currently, if you have pass a route into a `RootContainer` that has n queries, then n fetches are made to the server.

This actually seems like an issue that should be resolvable before the query even hits the network layer, so fixing it should mean good things even for custom network layers!

@yungsters @josephsavona @dschafer Is there any fundamental problem with just generating a union of the queries in `processQueue` before sending them off? Is this compatible with future plans for `@defer`'d queries?

**Comment by josephsavona** [comment]
Good question: there is no technical reason why multiple queries can't be grouped together; in fact, our internal network layer sends multiple queries in a single request. We simply didn't have time to implement this for OSS before the preview release. This is on the list of upcoming improvements, along with deferred queries.

**Comment by devknoll** [comment]
Actually, putting a little more thought into this, it might not be as trivial as just always generating a union of the queries. At least on a desktop with a fast internet connection, you might actually get improved performance by hitting multiple endpoint servers in parallel rather than waiting for a single one to complete all of the work.

**Comment by josephsavona** [comment]
A union is just one option, but it definitely is more performant to batch queries in a single request. I'm going to close this issue, but we'll continue to improve query batch going forward.
